tokio/src/task/scope/scope.rs

```diff
@@ -0,0 +1,354 @@
+use super::{
+    cancellation_token::CancellationToken,
+    wait_group::{WaitGroup, WaitGroupFuture},
+};
+use crate::task::{JoinError, JoinHandle};
+use pin_project_lite::pin_project;
+use std::{
+    future::Future,
+    pin::Pin,
+    sync::Arc,
+    task::{Context, Poll},
+};
+
+#[derive(Clone)]
+struct ScopeState {
+    cancel_token: Arc<CancellationToken>,
+    wait_group: SharedWaitGroup,
+    options: ScopeOptions,
+}
+
+impl ScopeState {
+    fn new(options: ScopeOptions) -> Self {
+        Self {
+            cancel_token: Arc::new(CancellationToken::new(false)),
+            wait_group: SharedWaitGroup::new(),
+            options,
+        }
+    }
+
+    fn is_cancelled(&self) -> bool {
+        self.cancel_token.is_cancelled()
+    }
+}
+
+pin_project! {
+    /// Allows to wait for a child task to join
+    pub struct ScopedJoinHandle<'scope, T> {
+        #[pin]
+        handle: JoinHandle<Result<T, CancellableFutureError>>,
+        phantom: core::marker::PhantomData<&'scope ()>,
```

Comment:
I think that, since the phantomdata is unused, this should have a leading `_`?

Reply:
indeed!